src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const currentState = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(stateCopy, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach(function (key) {
+        delete stateCopy[key];
+      });
+    }
+
+    if (action.type === 'clear') {
+      for (const key in stateCopy) {
+        if (stateCopy.hasOwnProperty(key)) {
+          delete stateCopy[key];
+        }
+      }
+    }
```

Comment:
[CODE STYLE]: Use switch statement if you have limited amount of conditions. Instead of using multiple `if` statements, you should use a `switch` statement to handle different action types. This will make your code cleaner and easier to understand.